test/ts/no-shadow-root.tsx

```diff
@@ -0,0 +1,123 @@
+import {
+  define,
+  Renderer,
+  WithRenderer,
+  WithUpdate,
+  withComponent,
+  withRenderer,
+  withChildren,
+  withUnique,
+  withUpdate,
+  WithComponent,
+  Constructor,
+  CustomElement
+} from 'skatejs';
+
+// custom class definition needed to get Generics support for custom mixin composition
+interface CustomComponentBase<P = {}, S = {}>
```

Comment:
same technique is used for defining `WithComponent` class which allows us to use generics to provide props/state/context

This is how user can provide their custom Base class strong type annotations for custom mixins composition